mps-model-adapters/src/main/kotlin/org/modelix/model/mpsadapters/MPSRepositoryAsNode.kt

```diff
@@ -78,34 +82,40 @@ data class MPSRepositoryAsNode(val repository: SRepository) : IDeprecatedNodeDef
     }
 
     override fun getReferenceTarget(link: IReferenceLink): INode? {
-        TODO("Not yet implemented")
+        return null
     }
 
     override fun setReferenceTarget(link: IReferenceLink, target: INode?) {
-        TODO("Not yet implemented")
+        if (target != null) {
+            throw IllegalArgumentException("$concept doesn't contain a reference link $link")
+        }
     }
 
     override fun setReferenceTarget(role: IReferenceLink, target: INodeReference?) {
-        TODO("Not yet implemented")
+        if (target != null) {
+            throw IllegalArgumentException("$concept doesn't contain a reference link $role")
+        }
     }
 
     override fun getReferenceTargetRef(role: IReferenceLink): INodeReference? {
-        TODO("Not yet implemented")
+        return null
     }
 
     override fun getPropertyValue(property: IProperty): String? {
-        TODO("Not yet implemented")
+        return null
     }
 
     override fun setPropertyValue(property: IProperty, value: String?) {
-        TODO("Not yet implemented")
+        if (value != null) {
+            throw IllegalArgumentException("$concept doesn't contain a property $property")
+        }
```

Comment:
Why is setting a `null` value to any non-existing property valid? I would have expected the same exception or an comment explaining the strange case.

Reply:
Because only then its symmetric to `getPropertyValue`. It would be unexpected for setPropertyValue to throw an exception when writing `setPropertyValue(property, getPropertyValue(property))`.

Reply:
That means there's no chance to disambiguate between an unknown property and a known one without a value from the reaction of `getPropertyValue`? Is that a good thing?

Reply:
INode is not responsible for enforcing the conformance to the meta model. Usually, implementations of INode are generic and store property values in a simple map. This MPSRepositoryAsNode is just a special case that cannot store any properties. Maybe the error message is a bit misleading. It would be legal to set a property that is not defined in the concept, but the implementation is just not capable of storing it.